docs/checkit.js

```diff
@@ -130,6 +130,128 @@ Calendar.prototype.generateMonthObjects = function (startDate, endDate) {
+};
+
+CalendarAnalyzer.prototype.getCheckedDaysStreak = function () {
+    // Returns the longest streak of checked days
+    "use strict";
+    // if checkedDays is an empty object, return 0;
+    if (jQuery.isEmptyObject(this.calState.checkedDays)) {
+        return 0;
+    }
+    
+    var currentStreak = [],
+        arrayOfStreakArrays = [],
+        checkedDaysArray = [],
+        key,
+        lastDate,
+        checkedDaysArrayLength,
+        i,
+        currentDay,
+        previousDay,
+        dayDiff,
+        largestArrayIndex;
+    
+    // place checkedDays object's keys into array and sort the array.
+    for (key in this.calState.checkedDays) {
+        if (this.calState.checkedDays.hasOwnProperty(key)) {
+            checkedDaysArray.push(key);
+        }
+    }
+    checkedDaysArray.sort();
+
+    checkedDaysArrayLength = checkedDaysArray.length;
+    
+    // for each checked day
+    for (i = 0; i < checkedDaysArrayLength; i++) {
+        
+        if (i === 0) {
+            lastDate = checkedDaysArray[i];
+            currentStreak.push(lastDate);
+            i++;
+        }
+        
+        // if the next element is the day after lastDate, add it to currentStreak
+        // if not, then add currentStreak to arrayOfStreakArrays.
+        currentDay = moment(checkedDaysArray[i], "YYYYMMDD");
+        previousDay = moment(checkedDaysArray[i - 1], "YYYYMMDD");
+        
+        dayDiff = currentDay.diff(previousDay, 'days');
+
+        if (currentDay.diff(previousDay, 'days') === 1) {
+            currentStreak.push(checkedDaysArray[i]);
+        } else {
+            arrayOfStreakArrays.push(currentStreak);
+            currentStreak = [];
+            currentStreak.push(checkedDaysArray[i]);
+        }
+        
+        // if on last element, push the currentStreak into the array of streaks
+        if (i === checkedDaysArrayLength - 1) {
+            arrayOfStreakArrays.push(currentStreak);
+        }
+        
+    }
+    
+    // Find the array with the most elements in the arrayOfStreakArrays
+    largestArrayIndex = arrayOfStreakArrays.reduce(function (maxI, el, i, arr) {
+        return el.length > arr[maxI].length ? i : maxI;
+    }, 0);
+   
+    return arrayOfStreakArrays[largestArrayIndex].length;
+};
+
+CalendarAnalyzer.prototype.getUncheckedDaysStreak = function () {
+    // Returns the longest streak of unchecked days
+    "use strict";
+    // Probably don't need this right?
+};
+
+CalendarAnalyzer.prototype.getTotalCalendarWeeks = function () {
+    // Returns the total number of active weeks in the calendar
+    "use strict";
+    var endDate = moment(this.calState.endDateString, "YYYYMMDD"),
+        startDate = moment(this.calState.startDateString, "YYYYMMDD"),
+        totalWeeks = endDate.diff(startDate, 'weeks');
+    return totalWeeks;
+};
+
+CalendarAnalyzer.prototype.getNumOfDaysLeft = function () {
+    // Returns the total number of active days left in a calendar starting from current
```

Comment:
active days... like... you could start your calendar on January 5th and end it on January 31st, then Jan 1st - 4th are inactive days, and the rest of the days in January are active.